common/persistence/sql/sqlplugin/postgresql/session/session.go

```diff
@@ -79,9 +83,24 @@ func createConnection(
 	d driver.Driver,
 	resolver resolver.ServiceResolver,
 ) (*sqlx.DB, error) {
-	db, err := d.CreateConnection(buildDSN(cfg, resolver))
-	if err != nil {
-		return nil, err
+	var db *sqlx.DB
+	if connector := cfg.DBConnector; connector != nil {
+		var driverName string
+		switch (*connector).Driver().(type) {
+		case *pgx.Driver:
+			driverName = "pgx"
+		case *pq.Driver:
+			driverName = "postgres"
+		default:
+			return nil, errors.New("invalid postgres driver")
+		}
+		db = sqlx.NewDb(sql.OpenDB(*cfg.DBConnector), driverName)
```

Comment:
Please pass the driver's name as part of the `config.SQL` object

Reply:
https://github.com/temporalio/temporal/pull/5265/commits/5348ec70077935f5907844d3d859e3273d3f4732